src/Gtk/Avalonia.Gtk/PixbufFramebuffer.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Avalonia.Controls.Platform.Surfaces;
+using Avalonia.Platform;
+using Gdk;
+
+namespace Avalonia.Gtk
+{
+    class PixbufFramebuffer : ILockedFramebuffer
+    {
+        private Pixbuf _pixbuf;
+        private Drawable _drawable;
+
+        public PixbufFramebuffer(int width, int height)
+        {
+            _pixbuf = new Pixbuf(Gdk.Colorspace.Rgb, false, 8, width, height);
+        }
+
+        public void SetDrawable(Drawable drawable)
```

Comment:
Backing surface is reused between `OnExpose` calls as long as size is the same, while Gdk.Drawable might change (in case of embedding for example).